booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -331,7 +331,7 @@ applyRule pat@Pattern{ceilConditions} rule = withRuleContext rule $ runRewriteRu
                 "Known true side conditions (won't check):" <+> pretty knownTrue
 
     unclearRequires <-
-        catMaybes <$> mapM (checkConstraint id notAppliedIfBottom) toCheck
+        catMaybes <$> mapM (checkConstraint id notAppliedIfBottom (Set.fromList knownTrue)) toCheck
```

Comment:
`knownTrue` are just the requires clauses from the rule that syntactically occur in the path condition after substituting. 
I think we want the _entire_ path condition (`pat.constraints`) instead here, and it comes from the subject, so no substitutions are required.
```suggestion
        catMaybes <$> mapM (checkConstraint id notAppliedIfBottom prior) toCheck
```
This means the argument can actually be _fixed_ in the local `checkConstraint` definition but we can also keep them as an argument for clarity.

Reply:
:facepalm: